pyesbulk/__init__.py

```diff
@@ -157,21 +233,43 @@ def actions_tracking_closure(cl_actions):
     generator = actions_tracking_closure(actions)
 
     streaming_bulk_generator = helpers.streaming_bulk(
-            es, generator, raise_on_error=False,
-            raise_on_exception=False, request_timeout=_request_timeout)
+        es,
+        generator,
+        raise_on_error=False,
+        raise_on_exception=False,
+        request_timeout=_request_timeout,
+    )
 
     for ok, resp_payload in streaming_bulk_generator:
         retry_count, action = actions_deque.popleft()
         try:
             resp = resp_payload[_op_type]
-            status = resp['status']
         except KeyError as e:
-            assert not ok
-            # resp is not of expected form
-            print(resp)
+            assert not ok, f"ok = {ok!r}, e = {e!r}"
+            assert (
+                e.args[0] == _op_type
+            ), f"e.args = {e.args!r}, _op_type = {_op_type!r}"
+            # For whatever reason, some errors are always returned using
+            # the "index" operation type instead of _op_type (e.g. "create"
+            # op type still comes back as an "index" response).
+            try:
+                resp = resp_payload["index"]
+            except KeyError:
+                # resp is not of expected form; set it to the complete
+                # payload, so that it can be reported properly below.
+                resp = resp_payload
+        try:
+            status = resp["status"]
+        except KeyError as e:
+            assert not ok, f"ok = {ok!r}, e = {e!r}"
+            # Limit the length of the error message.
```

Comment:
Is this a FIXME comment? I don't see how the length of the message is limited.

Reply:
Fixed.